tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/server/ConfigurationTest.java

```diff
@@ -122,6 +124,9 @@ public void testPropertiesConfigReload() throws Exception {
             return mytopic_31.getBindings().size() == 3;
          });
 
+         // verify round trip apply
+         Assert.assertTrue(server.getActiveMQServerControl().getStatus().contains("2"));
```

Comment:
Is the config update application atomic or can updates to one element be seen mid-application of others? If the latter then this doesnt necessarily seem to reliably achieve what you want as the 'status' value may be visible before other changes in the update are actually applied? I guess you could apply a second status-only change to the config, after the first round is seen, to know the first update process had finished, since you can see the second update in effect.

Reply:
they are atomic enough via the configuration reload. A new instance is created/modified and swapped. What I want to know in this case is that a reload has happened.

Reply:
I peeked a little more, and the reload does not use a new config object, so the status update is visible before the update. If we want better feedback in the future as to what was updated, once the changes were detected, it could be coordinated and communicated via the status. For now, with properties in a config map projection, I am happy that the status can reflect the refresh or the projection and the reaction of the reload task.